src/MainApp.js

```diff
@@ -0,0 +1,28 @@
+import React, { useContext } from "react";
+import Main from "./components/Main";
+import Sidebar from "./components/Sidebar";
+import Navi from "./components/Navi";
+import NoteDisplay from "./components/NoteDisplay/";
+import { ThemeContext } from "./contexts/ThemeContext";
+import NoteInfo from "./components/NoteInfo";
+import { AppContext } from "./contexts/AppContext";
+
+export default function MainApp() {
+    const { darkMode } = useContext(ThemeContext);
+    const { state } = useContext(AppContext);
+
+    return (
+        <React.Fragment>
+            <Main
+                darkMode={darkMode}
+                showInfo={state.infoBarOpen}
+                sidebarOpen={state.sidebarOpen}
+            >
+                <Navi/>
+                <Sidebar sidebarOpen={state.sidebarOpen}/>
```

Comment:
Since you are using the Context API, I think props drilling isn't necessary. Since this is what the Context API aims to solve. I proposed removing the props drilling for the `useContext` and use it within the components instead. Like adding a `useContext` inside the sidebar. Instead of using it in the parent component then passing it as props to its children. 😀